core/payment-driver/gnt/src/processor.rs

```diff
@@ -6,70 +8,141 @@ use std::sync::Arc;
 use ya_client_model::payment::Allocation;
 use ya_client_model::NodeId;
 use ya_core_model::driver::{AccountMode, PaymentConfirmation, PaymentDetails};
+use ya_persistence::executor::DbExecutor;
 
 #[derive(Clone)]
 pub struct GNTDriverProcessor {
-    driver: Arc<GntDriver>,
+    rinkeby_driver: Arc<GntDriver>,
+    mainnet_driver: Arc<GntDriver>,
 }
 
 impl GNTDriverProcessor {
-    pub fn new(driver: GntDriver) -> Self {
-        Self {
-            driver: Arc::new(driver),
-        }
+    pub async fn new(db: DbExecutor) -> GNTDriverResult<Self> {
+        Ok(Self {
+            rinkeby_driver: Arc::new(
+                GntDriver::new(db.clone(), Network::Rinkeby, *RINKEBY_CONFIG).await?,
+            ),
+            mainnet_driver: Arc::new(
+                GntDriver::new(db.clone(), Network::Mainnet, *MAINNET_CONFIG).await?,
+            ),
+        })
     }
 
     pub async fn account_locked(&self, identity: NodeId) -> GNTDriverResult<()> {
-        self.driver.account_locked(identity).await
+        self.rinkeby_driver.account_locked(identity).await?;
+        self.mainnet_driver.account_locked(identity).await
     }
 
     pub async fn account_unlocked(&self, identity: NodeId) -> GNTDriverResult<()> {
-        self.driver.account_unlocked(identity).await
+        self.rinkeby_driver.account_unlocked(identity).await?;
+        self.mainnet_driver.account_unlocked(identity).await
     }
 
-    pub async fn init(&self, mode: AccountMode, address: &str) -> GNTDriverResult<()> {
-        self.driver.init(mode, address).await
+    pub async fn fund(&self, address: &str, network: Network) -> GNTDriverResult<String> {
+        match network {
+            Network::Rinkeby => { self.rinkeby_driver.fund(address).await },
+            Network::Mainnet => { Ok(format!(
+                "Your mainnet ethereum address is {}. Send some GLM tokens and ETH for gas to this address \
+                to be able to use this driver. Using this driver is not recommended. If you want to easily \
+                acquire some GLM to try Golem on mainnet please use zksync driver.", address
+            )) }
+        }
     }
 
-    pub async fn get_account_balance(&self, address: &str) -> GNTDriverResult<BigDecimal> {
-        self.driver.get_account_balance(address).await
+    pub async fn init(
+        &self,
+        mode: AccountMode,
+        address: &str,
+        network: Network,
+    ) -> GNTDriverResult<()> {
+        match network {
+            Network::Rinkeby => self.rinkeby_driver.init(mode, address).await,
+            Network::Mainnet => self.mainnet_driver.init(mode, address).await,
+        }
+    }
+
+    pub async fn get_account_balance(
+        &self,
+        address: &str,
+        network: Network,
+    ) -> GNTDriverResult<BigDecimal> {
+        match network {
```

Comment:
this `match network`block is repeated a lot, could be something lie `self.get_network_driver(network).do_function().await`

Reply:
I'll give that a try.